util/webhook/webhook.go

```diff
@@ -132,13 +135,24 @@ func (a *ArgoCDWebhookHandler) startWorkerPool(webhookParallelism int) {
 }
 
 func ParseRevision(ref string) string {
+	if ref == "HEAD" {
+		return ref
+	}
 	refParts := strings.SplitN(ref, "/", 3)
 	return refParts[len(refParts)-1]
 }
 
 // affectedRevisionInfo examines a payload from a webhook event, and extracts the repo web URL,
 // the revision, and whether or not this affected origin/HEAD (the default branch of the repository)
-func affectedRevisionInfo(payloadIf interface{}) (webURLs []string, revision string, change changeInfo, touchedHead bool, changedFiles []string) {
+func affectedRevisionInfo(payloadIf interface{}, optionalHandler ...interface{}) (webURLs []string, revision string, change changeInfo, touchedHead bool, changedFiles []string) {
+	// Check if a handler was passed
+	var webhookHandler *ArgoCDWebhookHandler
+	if len(optionalHandler) > 0 {
+		if handler, ok := optionalHandler[0].(*ArgoCDWebhookHandler); ok {
```

Comment:
This leads to a confusing function signature, which implies multiple optional handlers can be passed. Maybe use a pointer and have it be nil if the optional handler is not available?

Reply:
fixed pls check